### PR TITLE
ANW-625: Refactor staff drag and drop modal as toolbar buttons

### PR DESCRIPTION
**Pull request by brianzelip**
## Description

[ANW-625](https://archivesspace.atlassian.net/browse/ANW-625): Refactor the UI for asking the user "how do you want to handle dropping your dragged selection", from a post-drop modal to a pre-drag radio input selection with a default choice.

Features of the new radio input buttons include:
- built from semantic HTML, and therefore are more accessible and operate slightly differently under the hood than other buttons while remaining visually and functionally consistent with the other toolbar buttons
- make use of the browser's local storage to remember the user's most recent selection

## ⚠️ Docs update required ⚠️ 
**NOTE** that this change requires updating the docs here, https://archivesspace.atlassian.net/wiki/spaces/ArchivesSpaceUserManual/pages/894140803/Reordering+Components+in+Resources+and+Digital+Objects+as+of+v2.0.0

### Before

ANW-625-before

### New drag and drop in action

ANW-625-drag-drop

### Keyboard navigation of buttons, showing browser's local storage value

ANW-625-keyboard-storage

## How Has This Been Tested?

See frontend/spec/selenium/spec/digital_objects_spec.rb

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have authority to submit this code.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

[ANW-625]: https://archivesspace.atlassian.net/browse/ANW-625?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by quoideneuf — CHANGES_REQUESTED** [review]
@brianzelip I only found one issue with this, which is that if the user does not select an option in the toolbar, `dropBehavior` will remain `undefined` and the `handleDrop` function will silently miss each of the `switch` options.

File: frontend/app/assets/javascripts/largetree_dragdrop.js.erb [hunk 1]
```diff
@@ -9,6 +9,7 @@
         this.dragActive = false;
         this.dragIndicator = $('<div class="tree-drag-indicator" />');
         this.rowsToMove = [];
+        this.dropBehavior = undefined;
```
Comment (quoideneuf) [inline frontend/app/assets/javascripts/largetree_dragdrop.js.erb:12]: Can this just be initialized to `"before"` to match the default in the toolbar?

Comment (brianzelip) [reply frontend/app/assets/javascripts/largetree_dragdrop.js.erb]: Basically; had to also consider local storage.

**Comment by brianzelip** [comment]
@quoideneuf Good and simple catch! I was so focused on the new toolbar buttons that I overlooked page load 🤦

**Pull request merged**